pkg/cli/client/logs/logs.go

```diff
@@ -0,0 +1,51 @@
+package logs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/tektoncd/results/pkg/cli/client"
+	pb "github.com/tektoncd/results/proto/v1alpha2/results_go_proto"
+)
+
+// Client is a client for interacting with logs.
+type Client struct {
+	*client.RESTClient
+}
+
+// NewClient creates a new logs client.
+func NewClient(rc *client.RESTClient) *Client {
+	return &Client{RESTClient: rc}
+}
+
+// GetLog gets a log by name.
+func (c *Client) GetLog(ctx context.Context, req *pb.GetLogRequest) (io.Reader, error) {
+	// Create a pipe to stream the logs
+	pr, pw := io.Pipe()
+
+	// Start a goroutine to handle the streaming response
+	go func() {
+		defer pw.Close()
+
+		// Build the URL for the log request, replacing "records" with "logs" in the path
+		url := c.BuildURL(fmt.Sprintf("parents/%s", strings.Replace(req.Name, "records", "logs", 1)), nil)
+
+		// Make the request using the RESTClient's DoRequest method
+		out, err := c.DoRequest(ctx, http.MethodGet, url, nil, nil)
+		if err != nil {
```

Comment:
What's the reason we're modifying the `DoRequest` to return the response serialized, instead of leaving its implementation as-is and doing something like the following:
```go
response := &pb.GetLogResponse{} // or whatever shape we expect
err := c.DoRequest(ctx, http.MethodGet, url, nil, response)
if err != nil { ... }
output, err := protojson.Marshal(response)
if err != nil { ... }
pw.Write(output)
```

This is admittedly a little extra compute, since we're redundantly unmarshalling then marshalling, but I think it would make the `DoRequest` API a lot easier to reason about. WDYT?

Reply:
As we discussed, I have made the changes according to the approach explained above. 
Please take a look and let me know if any further changes are required.